examples/data-objects/smde/src/smdeView.tsx

```diff
@@ -0,0 +1,182 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IRequest } from "@fluidframework/core-interfaces";
+import { mixinRequestHandler } from "@fluidframework/datastore";
+import { SharedMap } from "@fluidframework/map";
+import {
+	MergeTreeDeltaType,
+	TextSegment,
+	ReferenceType,
+	reservedTileLabelsKey,
+	Marker,
+} from "@fluidframework/merge-tree";
+import {
+	IFluidDataStoreContext,
+	IFluidDataStoreFactory,
+} from "@fluidframework/runtime-definitions";
+import { getTextAndMarkers, SharedString } from "@fluidframework/sequence";
+
+import React, { useEffect, useRef } from "react";
+import SimpleMDE from "simplemde";
+
+import { SmdeDataObject } from "./smde";
+
+// eslint-disable-next-line import/no-internal-modules, import/no-unassigned-import
+import "simplemde/dist/simplemde.min.css";
+
+class SmdeView {
+	private textArea: HTMLTextAreaElement | undefined;
+	private smde: SimpleMDE | undefined;
+
+	public constructor(private readonly smdeDataObject: SmdeDataObject) {}
+
+	public render(elm: HTMLElement): void {
+		// Create base textarea
+		if (!this.textArea) {
+			this.textArea = document.createElement("textarea");
+		}
+
+		// Reparent if needed
+		if (this.textArea.parentElement !== elm) {
+			this.textArea.remove();
+			elm.appendChild(this.textArea);
+		}
+
+		if (!this.smde) {
+			this.setupEditor();
+		}
+	}
+
+	private setupEditor() {
+		const smde = new SimpleMDE({ element: this.textArea });
+		this.smde = smde;
+
+		const { parallelText } = getTextAndMarkers(this.smdeDataObject.text, "pg");
+		const text = parallelText.join("\n");
+		this.smde.value(text);
+
+		let localEdit = false;
+
+		this.smdeDataObject.text.on("sequenceDelta", (ev) => {
+			if (ev.isLocal) {
```

Comment:
Nit: probably worth a brief comment as to why we want to early-out for local events.

Reply:
Done in latest